common/src/main/java/com/wynntils/screens/maps/PoiManagementScreen.java

```diff
@@ -0,0 +1,243 @@
+        if (deleteCount <= 0) {
+            deleteAllButton.setMessage(Component.literal("DELETED!"));
+        }
+
+        waypoints =
+                Managers.Feature.getFeatureInstance(MapFeature.class).customPois.get();
+
+        if (!waypoints.isEmpty() && page * pageHeight > waypoints.size() - 1) {
+            page = (waypoints.size() - 1) / pageHeight;
+        }
+
+        checkAvailablePages();
+
+        populatePois();
+    }
+
+    @Override
+    public void doRender(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        renderBackground(poseStack);
+        super.doRender(poseStack, mouseX, mouseY, partialTick);
+
+        FontRenderer.getInstance()
+                .renderText(
+                        poseStack,
+                        I18n.get("screens.wynntils.poiManagementGui.icon"),
+                        this.width / 2f - 165,
+                        43,
+                        CommonColors.WHITE,
+                        HorizontalAlignment.Left,
+                        VerticalAlignment.Top,
+                        TextShadow.NORMAL);
+
+        FontRenderer.getInstance()
+                .renderText(
+                        poseStack,
+                        I18n.get("screens.wynntils.poiManagementGui.name"),
+                        this.width / 2f - 130,
+                        43,
+                        CommonColors.WHITE,
+                        HorizontalAlignment.Left,
+                        VerticalAlignment.Top,
+                        TextShadow.NORMAL);
+
+        FontRenderer.getInstance()
+                .renderText(
+                        poseStack,
+                        "X",
+                        this.width / 2f - 15,
+                        43,
+                        CommonColors.WHITE,
+                        HorizontalAlignment.Left,
+                        VerticalAlignment.Top,
+                        TextShadow.NORMAL);
+
+        FontRenderer.getInstance()
+                .renderText(
+                        poseStack,
+                        "Y",
+                        this.width / 2f + 40,
+                        43,
+                        CommonColors.WHITE,
+                        HorizontalAlignment.Left,
+                        VerticalAlignment.Top,
+                        TextShadow.NORMAL);
+
+        FontRenderer.getInstance()
+                .renderText(
+                        poseStack,
+                        "Z",
+                        this.width / 2f + 80,
+                        43,
+                        CommonColors.WHITE,
+                        HorizontalAlignment.Left,
+                        VerticalAlignment.Top,
+                        TextShadow.NORMAL);
+
+        RenderUtils.drawRect(poseStack, CommonColors.WHITE, this.width / 2 - 165, 52, 0, 355, 1);
+
+        poiManagerWidgets.forEach(widget -> widget.render(poseStack, mouseX, mouseY, partialTick));
```

Comment:
You don't need this as you call super.render.

Reply:
I've tried that but it seems to make some icons randomly dissappear 

https://user-images.githubusercontent.com/8337467/228920141-9e7d07c0-902a-419e-9846-34b7bfa838bb.mp4

Reply:
Then, there is something really wrong..

Reply:
Yes, that is worrisome. Overall, the PR looks pretty okay now, but the root cause of the disappearing icons must be found.